pkg/query-service/telemetry/telemetry.go

```diff
@@ -74,28 +75,32 @@ func (a *Telemetry) IsSampled() bool {
 
 }
 
-func (telemetry *Telemetry) CheckSigNozMetricsV2(compositeQuery *model.CompositeMetricQuery) bool {
+func (telemetry *Telemetry) CheckSigNozSignals(postData *v3.QueryRangeParamsV3) (bool, bool) {
+	signozLogsUsed := false
+	signozMetricsUsed := false
 
-	signozMetricsNotFound := false
-
-	if compositeQuery.BuilderQueries != nil && len(compositeQuery.BuilderQueries) > 0 {
-		if !strings.Contains(compositeQuery.BuilderQueries["A"].MetricName, "signoz_") && len(compositeQuery.BuilderQueries["A"].MetricName) > 0 {
-			signozMetricsNotFound = true
+	if postData.CompositeQuery.QueryType == v3.QueryTypeBuilder {
+		for _, query := range postData.CompositeQuery.BuilderQueries {
+			if query.DataSource == v3.DataSourceLogs && len(query.Filters.Items) > 0 {
+				signozLogsUsed = true
+			} else if query.DataSource == v3.DataSourceMetrics && !strings.Contains(query.AggregateAttribute.Key, "signoz_") {
+				signozMetricsUsed = true
+			}
 		}
-	}
-
-	if compositeQuery.PromQueries != nil && len(compositeQuery.PromQueries) > 0 {
-		if !strings.Contains(compositeQuery.PromQueries["A"].Query, "signoz_") && len(compositeQuery.PromQueries["A"].Query) > 0 {
-			signozMetricsNotFound = true
+	} else if postData.CompositeQuery.QueryType == v3.QueryTypePromQL {
+		for _, query := range postData.CompositeQuery.PromQueries {
+			if !strings.Contains(query.Query, "signoz_") && len(query.Query) > 0 {
+				signozMetricsUsed = true
+			}
 		}
-	}
-	if compositeQuery.ClickHouseQueries != nil && len(compositeQuery.ClickHouseQueries) > 0 {
-		if !strings.Contains(compositeQuery.ClickHouseQueries["A"].Query, "signoz_") && len(compositeQuery.ClickHouseQueries["A"].Query) > 0 {
-			signozMetricsNotFound = true
+	} else if postData.CompositeQuery.QueryType == v3.QueryTypeClickHouseSQL {
+		for _, query := range postData.CompositeQuery.ClickHouseQueries {
+			if !strings.Contains(query.Query, "signoz_metrics") && len(query.Query) > 0 {
```

Comment:
Should it be `signoz_` instead of `signoz_metrics`?

Reply:
We are not using Clickhouse query in default dashboard so we can actually consider any clickhouse query which uses `signoz_metrics` table as query written by user.
The condition should be updated to `strings.Contains(query.Query, "signoz_metrics")` instead of `!strings.Contains....`

Reply:
We used to reject events caused by APM metrics which contain `signoz_` in metrics name. How is that handled here?

Reply:
That is handled here: https://github.com/SigNoz/signoz/blob/f36e16084c9acb8f22466585d513ad97d8ec06fc/pkg/query-service/telemetry/telemetry.go#L86

Reply:
shouldn't this check be added to ClickhouseBuilder too? Or you want to consider folks users creating CH queries for APM to be real metrics users. It might be okay to make that assumption.

Reply:
Yeah we are making that assumption as we don't have any examples to write clickhouse query on metrics in our docs and it's very unlikely a novice user will be able to perform this action.

Reply:
Yes, users will add variables only if data is already there. So we can consider user as active user of metrics.